src/coreclr/jit/valuenum.cpp

```diff
@@ -9584,7 +9598,25 @@ ValueNum ValueNumStore::VNForBitCast(ValueNum srcVN, var_types castToType)
         return VNZeroForType(castToType);
     }
 
-    return VNForFunc(castToType, VNF_BitCast, srcVN, VNForIntCon(castToType));
+    ValueNum srcExcVN;
+    ValueNum srcNormVN;
+    VNUnpackExc(srcVN, &srcNormVN, &srcExcVN);
+
+    ValueNum resultNormVN = VNForFunc(castToType, VNF_BitCast, srcNormVN, VNForIntCon(castToType));
+    ValueNum resultExcVN  = srcExcVN;
+    
+    return VNWithExc(resultNormVN, resultExcVN);
+}
+
+// Compute the ValueNumberPair for a bitcast operation
+ValueNumPair ValueNumStore::VNPairForBitCast(ValueNumPair srcVNPair, var_types castToType)
+{
+    ValueNum srcLibVN     = srcVNPair.GetLiberal();
+    ValueNum srcConVN     = srcVNPair.GetConservative();
+    ValueNum bitCastLibVN = VNForBitCast(srcLibVN, castToType);
+    ValueNum bitCastConVN = VNForBitCast(srcConVN, castToType);
+
+    return {bitCastLibVN, bitCastConVN};
```

Comment:
Nit: I realize `VNPairForCast` is doing this, but the general style in VN is to call the constructor explicitly: `return ValueNumPair(bitCastLibVN, bitCastConVN)`.

(See e. g. the various methods that deal with exception sets)

Reply:
Fixed.